From 8d9179562ea961226e9467ccdb80ed2d9c41707a Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 11 Jan 2024 10:25:19 +0100
Subject: [PATCH 0448/1695] dts: stm32mp25: Define scmi regulator consumer

Define voltage regulator exposed by the SCMI server using the device
tree compatible node st,scmi-regulator-consumer for STM32MP25 platforms.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 408fe16dec6fb2dd51393ee6d0cf7098fe477d64 dts: stm32mp25: Define scmi regulator consumer]
Change-Id: I0bb521b3f3fbaab06a3b0d3dc38400d907d9496d
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi             | 38 +++++++++++++++
 core/arch/arm/dts/stm32mp257f-dk.dts          | 32 +++++++++++++
 core/arch/arm/dts/stm32mp257f-ev1.dts         | 16 +++++++
 .../regulator/st,stm32mp25-regulator.h        | 48 +++++++++++++++++++
 4 files changed, 134 insertions(+)
 create mode 100644 core/include/dt-bindings/regulator/st,stm32mp25-regulator.h

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 6fcf5b9f1..adfc1e482 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -6,6 +6,7 @@
 
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/clock/stm32mp25-clks.h>
+#include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/reset/stm32mp25-resets.h>
 
 / {
@@ -84,6 +85,43 @@
 		};
 	};
 
+	scmi_regu: scmi-regu {
+		compatible = "st,scmi-regulator-consumer";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		scmi-channel-id = <0>;
+
+		voltd-vddio1 {
+			reg = <VOLTD_SCMI_VDDIO1>;
+			voltd-supply = <&vddio1>;
+		};
+		voltd-vddio2 {
+			reg = <VOLTD_SCMI_VDDIO2>;
+			voltd-supply = <&vddio2>;
+		};
+		voltd-vddio3 {
+			reg = <VOLTD_SCMI_VDDIO3>;
+			voltd-supply = <&vddio3>;
+		};
+		voltd-vddio4 {
+			reg = <VOLTD_SCMI_VDDIO4>;
+			voltd-supply = <&vddio4>;
+		};
+		voltd-vdd33ucpd {
+			reg = <VOLTD_SCMI_UCPD>;
+			voltd-supply = <&vdd33ucpd>;
+		};
+		voltd-vdda18adc {
+			reg = <VOLTD_SCMI_ADC>;
+			voltd-supply = <&vdda18adc>;
+		};
+		voltd-vddgpu {
+			reg = <VOLTD_SCMI_GPU>;
+			voltd-supply = <&vddgpu>;
+		};
+	};
+
 	soc@0 {
 		compatible = "simple-bus";
 		#address-cells = <1>;
diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index beb031cca..d5eec7937 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -6,6 +6,7 @@
 
 /dts-v1/;
 
+#include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/tamper/st,stm32mp25-tamp.h>
 #include "stm32mp257.dtsi"
@@ -389,6 +390,37 @@
 	status = "okay";
 };
 
+&scmi_regu {
+	voltd-v1v8 {
+		reg = <VOLTD_SCMI_STPMIC2_BUCK5>;
+		voltd-supply = <&v1v8>;
+	};
+	voltd-v3v3 {
+		reg = <VOLTD_SCMI_STPMIC2_BUCK7>;
+		voltd-supply = <&v3v3>;
+	};
+	voltd-v5v-hdmi {
+		reg = <VOLTD_SCMI_STPMIC2_LDO5>;
+		voltd-supply = <&v5v_hdmi>;
+	};
+	voltd-v5v-vconn {
+		reg = <VOLTD_SCMI_STPMIC2_LDO6>;
+		voltd-supply = <&v5v_vconn>;
+	};
+	voltd-vdd-emmc {
+		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
+		voltd-supply = <&vdd_emmc>;
+	};
+	voltd-vdd3v3-usb {
+		reg = <VOLTD_SCMI_STPMIC2_LDO4>;
+		voltd-supply = <&vdd3v3_usb>;
+	};
+	voltd-vdd-sdcard {
+		reg = <VOLTD_SCMI_STPMIC2_LDO7>;
+		voltd-supply = <&vdd_sdcard>;
+	};
+};
+
 &tamp {
 	status = "okay";
 	st,tamp-passive-precharge = <2>;
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index e0fab41b2..83600f334 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -7,6 +7,7 @@
 /dts-v1/;
 
 #include <dt-bindings/clock/stm32mp25-clksrc.h>
+#include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/soc/stm32mp25-rif.h>
 #include <dt-bindings/soc/stm32mp25-risab.h>
@@ -430,6 +431,21 @@
 	status = "okay";
 };
 
+&scmi_regu {
+	scmi_v3v3: voltd-v3v3 {
+		reg = <VOLTD_SCMI_STPMIC2_BUCK7>;
+		voltd-supply = <&v3v3>;
+	};
+	scmi_vdd_emmc: voltd-vdd-emmc {
+		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
+		voltd-supply = <&vdd_emmc>;
+	};
+	scmi_vdd_sdcard: voltd-vdd_sdcard {
+		reg = <VOLTD_SCMI_STPMIC2_LDO7>;
+		voltd-supply = <&vdd_sdcard>;
+	};
+};
+
 &tamp {
 	status = "okay";
 	st,tamp-passive-precharge = <2>;
diff --git a/core/include/dt-bindings/regulator/st,stm32mp25-regulator.h b/core/include/dt-bindings/regulator/st,stm32mp25-regulator.h
new file mode 100644
index 000000000..81d0ba479
--- /dev/null
+++ b/core/include/dt-bindings/regulator/st,stm32mp25-regulator.h
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-2-Clause) */
+/*
+ * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef __DT_BINDINGS_REGULATOR_STM32MP25_REGULATOR_H
+#define __DT_BINDINGS_REGULATOR_STM32MP25_REGULATOR_H
+
+/* SCMI voltage domains identifiers */
+
+/* SOC Internal regulators */
+#define VOLTD_SCMI_VDDIO1		0
+#define VOLTD_SCMI_VDDIO2		1
+#define VOLTD_SCMI_VDDIO3		2
+#define VOLTD_SCMI_VDDIO4		3
+#define VOLTD_SCMI_VDDIO		4
+#define VOLTD_SCMI_UCPD			5
+#define VOLTD_SCMI_USB33		6
+#define VOLTD_SCMI_ADC			7
+#define VOLTD_SCMI_GPU			8
+#define VOLTD_SCMI_VREFBUF		9
+
+/* STPMIC2 regulators */
+#define VOLTD_SCMI_STPMIC2_BUCK1	10
+#define VOLTD_SCMI_STPMIC2_BUCK2	11
+#define VOLTD_SCMI_STPMIC2_BUCK3	12
+#define VOLTD_SCMI_STPMIC2_BUCK4	13
+#define VOLTD_SCMI_STPMIC2_BUCK5	14
+#define VOLTD_SCMI_STPMIC2_BUCK6	15
+#define VOLTD_SCMI_STPMIC2_BUCK7	16
+#define VOLTD_SCMI_STPMIC2_LDO1		17
+#define VOLTD_SCMI_STPMIC2_LDO2		18
+#define VOLTD_SCMI_STPMIC2_LDO3		19
+#define VOLTD_SCMI_STPMIC2_LDO4		20
+#define VOLTD_SCMI_STPMIC2_LDO5		21
+#define VOLTD_SCMI_STPMIC2_LDO6		22
+#define VOLTD_SCMI_STPMIC2_LDO7		23
+#define VOLTD_SCMI_STPMIC2_LDO8		24
+#define VOLTD_SCMI_STPMIC2_REFDDR	25
+
+/* External regulators */
+#define VOLTD_SCMI_REGU0		26
+#define VOLTD_SCMI_REGU1		27
+#define VOLTD_SCMI_REGU2		28
+#define VOLTD_SCMI_REGU3		29
+#define VOLTD_SCMI_REGU4		30
+
+#endif /*__DT_BINDINGS_REGULATOR_STM32MP25_REGULATOR_H */
-- 
2.47.2

